samples/canopen/canopen_rpdo/main.cpp

```diff
@@ -0,0 +1,189 @@
+// EVT-core CAN callback and CAN setup. This will include logic to set
+// aside CANopen messages into a specific queue
+///////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Interrupt handler to get CAN messages. A function pointer to this function
+ * will be passed to the EVT-core CAN interface which will in turn call this
+ * function each time a new CAN message comes in.
+ *
+ * NOTE: For this sample, every non-extended (so 11 bit CAN IDs) will be
+ * assumed to be intended to be passed as a CANopen message.
+ *
+ * @param message[in] The passed in CAN message that was read.
+ */
+
+IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+
+// create a can interrupt handler
+void canInterrupt(IO::CANMessage& message, void* priv) {
+    EVT::core::types::FixedQueue<CANOPEN_QUEUE_SIZE, IO::CANMessage>* queue =
+        (EVT::core::types::FixedQueue<CANOPEN_QUEUE_SIZE, IO::CANMessage>*) priv;
+
+    //print out raw received data
+    uart.printf("Got RAW message from %X of length %d with data: ", message.getId(), message.getDataLength());
+    uint8_t* data = message.getPayload();
+    for (int i = 0; i < message.getDataLength(); i++) {
+        uart.printf("%X ", *data);
+        data++;
+    }
+    uart.printf("\r\n");
+
+    if (queue != nullptr)
+        queue->append(message);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// CANopen specific Callbacks. Need to be defined in some location
+///////////////////////////////////////////////////////////////////////////////
+extern "C" void CONodeFatalError(void) {}
+
+extern "C" void COIfCanReceive(CO_IF_FRM* frm) {}
+
+extern "C" int16_t COLssStore(uint32_t baudrate, uint8_t nodeId) { return 0; }
+
+extern "C" int16_t COLssLoad(uint32_t* baudrate, uint8_t* nodeId) { return 0; }
+
+extern "C" void CONmtModeChange(CO_NMT* nmt, CO_MODE mode) {}
+
+extern "C" void CONmtHbConsEvent(CO_NMT* nmt, uint8_t nodeId) {}
+
+extern "C" void CONmtHbConsChange(CO_NMT* nmt, uint8_t nodeId, CO_MODE mode) {}
+
+extern "C" int16_t COParaDefault(CO_PARA* pg) { return 0; }
+
+extern "C" void COPdoTransmit(CO_IF_FRM* frm) {}
+
+extern "C" int16_t COPdoReceive(CO_IF_FRM* frm) { return 0; }
+
+extern "C" void COPdoSyncUpdate(CO_RPDO* pdo) {}
+
+extern "C" void COTmrLock(void) {}
+
+extern "C" void COTmrUnlock(void) {}
+
+int main() {
+    // Initialize system
+    IO::init();
+
+    // Will store CANopen messages that will be populated by the EVT-core CAN
+    // interrupt
+    EVT::core::types::FixedQueue<CANOPEN_QUEUE_SIZE, IO::CANMessage> canOpenQueue;
+
+    // Initialize CAN, add an IRQ which will add messages to the queue above
+    IO::CAN& can = IO::getCAN<IO::Pin::PA_12, IO::Pin::PA_11>();
+    can.addIRQHandler(canInterrupt, reinterpret_cast<void*>(&canOpenQueue));
+
+    // Initialize the timer
+    DEV::Timerf302x8 timer(TIM2, 100);
+
+    // UART for testing
```

Comment:
Did you delete your UART initialization here?

Reply:
Looks like the UART initialization is done at the top of the file for some reason

Reply:
It is used in the event handles and I reused the same file as the TPDO sample with slight changes. The left over comment has been removed.